Metalama.Framework.Engine/GeneratedCodeAnalyzer.cs

```diff
@@ -37,12 +39,22 @@ public override void Initialize( AnalysisContext context )
 
     private void AnalyzeSymbol( SymbolAnalysisContext context )
     {
-#if ROSLYN_4_4_0_OR_GREATER // Roslyn 4.0 doesn't have IsGeneratedCode, so just do nothing in that case.
+        var tree = context.Symbol.GetClosestPrimaryDeclarationSyntax()?.SyntaxTree;
 
-        // IsGeneratedCode is based on heuristics, so it's not going to be exactly the same as files produced by source generators, but I can't think of a better way to do this.
-        if ( !context.IsGeneratedCode )
+        if ( MetalamaCompilerInfo.IsActive )
         {
-            return;
+            if ( tree == null || !SourceGeneratedCodeTracker.IsGenerated( tree ) )
```

Comment:
The code is slightly simplified if `IsGenerated` returns `null` when it does not know.